libwild/src/layout.rs

```diff
@@ -704,6 +704,24 @@ trait SymbolRequestHandler<'data>: std::fmt::Display + HandlerData {
                 }
             }
 
+            if symbol_db.args.got_plt_syms {
+                let name = symbol_db.symbol_name(symbol_id)?;
+                let mut name_len = name.len();
+                if current_res_flags.needs_got() || current_res_flags.needs_plt() {
+                    name_len += 4; // "$got" or "$plt" suffix
+                } else {
+                    continue;
```

Comment:
I'm not sure about this continue being here. It looks like it might skip stuff that it shouldn't - e.g. the call to `allocate_symbol_resolution` below. How about getting rid of the else-branch and just moving the allocation code into the if-branch? Actually, you could probably change your outer if-statement to `if symbol_db.args.got_plt_syms && current_res_flags.needs_got()`. This relies on `needs_got` always being set when `needs_plt` is set, but your code already relies on that below.

Reply:
I'd suggest adding some print statements to the code both at the point where you allocate and the point where you write. You can then run the linker outside of the test by copy and pasting the `cargo run` command from the test output. Hopefully that'll give you a clue as to why allocation and writing are inconsistent. If get stuck, let me know and I can help.